src/tools/perf/api/libperf.h

```diff
@@ -164,12 +172,20 @@ typedef struct ucx_perf_rte {
     ucx_perf_rte_recv_func_t         recv;
     ucx_perf_rte_exchange_vec_func_t exchange_vec;
 
-    /* Handle results */
-    ucx_perf_rte_report_func_t       report;
+    /* List of supported RTE */
+    ucs_list_link_t                  list;
 
 } ucx_perf_rte_t;
 
 
+/**
+ * Common report function
+ */
+typedef void (*ucx_perf_report_func_t)(void *rte_group,
+                                       const ucx_perf_result_t *result,
+                                       void *arg, const char *extra_info,
+                                       int is_final, int is_multi_thread);
```

Comment:
since this function is moved to perftest context now, maybe many params could be removed and we can pass perftest context instead (also remove report_arg)

Reply:
not sure, `struct perftest_context` contains `perftest_params_t params`, which has `report_func` and `report_args`. So libperf never sees `struct perftest_context`. if we really want that change, i'd suggest to keep it in another PR, as mad rte as a perftest module is independent from that.